core/src/main/java/org/testcontainers/containers/Container.java

```diff
@@ -287,6 +287,20 @@ default SELF withClasspathResourceMapping(final String resourcePath, final Strin
     Boolean isRunning();
 
     /**
+     * Get the actual mapped port for a first port exposed by the container.
+     *
+     * @return the port that the exposed port is mapped to
+     * @throws IllegalStateException if there are no exposed ports
+     */
+    default Integer getFirstMappedPort() {
+        return getExposedPorts()
```

Comment:
Well, at least I wouldn't say it's a misuse :)

Also, I prefer `findFirst + map` over `!isEmpty() + .get(0)`, less error prone if you ask me